### ISSUE TITLE
Application restart on trivial errors

### ISSUE THREAD

**Issue by yanickrochon** [keymetrics/pmx#30]
Since I enabled pmx on my application, to monitor error on keymetrics.io, the app restart on every trivial errors, which force users to login every time. Is there a way to *not* restart the application automatically on errors?

**Comment by Unitech** [comment]
An exception is never trivial! You should handle errors differently or wrap your code into a domain if you really want to keep your application alive after an expcetion

**Comment by jshkurti** [comment]
Are you saying it didn't before pmx ?
Show us an example please.
You can catch errors with process.on('uncaughtException') but it is not suggested.
You can also use domains : https://nodejs.org/api/domain.html

But in any case, pmx doesn't interfere with this behaviour.

**Comment by yanickrochon** [comment]
The errors happen in a very dark-aged piece of code within the bowel of this application. The piece of code was an attempt to synchronize clients using websockets and Primus, where clients could register to some event pattern and be notified (HTTP push), but the code was left untouched after random errors were thrown by the custom event emitter with obscure to unknown reason leading the event name to contain an undefined part, which case the emitter to throw on "Invalid event name".

The bottom line is that the error only affected websocket connectionss, and not the rest of the application. I wrapped the damn thing with a try...catch (with error logging, as a reminder to fix that crap at some point) and everything is fine since then.

BTW: I understand how errors should be handled, and how an app is better rebooting then continuing in an unstable state (mem leaks, wrong state, pending queries, etc.). But this does not apply here.
